### PR TITLE
Listmos speed improvements

### PR DESCRIPTION
**Pull request by neersighted**

### CODE REVIEW

File: code/modules/atmospherics/gasmixtures/gas_mixture.dm [hunk 1]
```diff
@@ -172,45 +171,46 @@ var/list/cached_gases_list = null
 				if(new_heat_capacity > MINIMUM_HEAT_CAPACITY)
 					temperature = max(((temperature*old_heat_capacity + reaction_energy)/new_heat_capacity),TCMB)
 					//Prevents whatever mechanism is causing it to hit negative temperatures.
-				//world << "post [temperature], [procgases["plasma"][MOLES]], [procgases["co2"][MOLES]]
+				//world << "post [temperature], [cached_gases["plasma"][MOLES]], [cached_gases["co2"][MOLES]]
```
Comment (duncathan) [inline code/modules/atmospherics/gasmixtures/gas_mixture.dm]: I love how we're still maintaining fusion

File: code/modules/atmospherics/gasmixtures/gas_mixture.dm [hunk 2]
```diff
@@ -337,41 +340,46 @@ var/list/cached_gases_list = null
...
-		cached_gases[id][MOLES] = 0
-
-	garbage_collect()
 
 	temperature = sample.temperature
+	for(var/id in sample_gases)
+		assert_gas(id)
+		cached_gases[id][MOLES] = sample_gases[id][MOLES]
+		copied_gases += id
+	cached_gases &= copied_gases
```
Comment (duncathan) [inline code/modules/atmospherics/gasmixtures/gas_mixture.dm]: I love this line

**Comment by neersighted** [comment]
Gotta go fast

File: code/modules/atmospherics/gasmixtures/gas_mixture.dm [hunk 3]
```diff
@@ -8,91 +8,89 @@ What are the archived variables for?
...
-
-	var/list/gas = cached_gases_list[gasid]
-	. = gas.Copy()
+	var/list/cached_gas = gaslist_cache[id]
+	return cached_gas.Copy()
 
 /datum/gas_mixture
 	var/list/gases
-	var/temperature //in Kelvin
+	var/temperature // degrees Kelvin
```
Comment (duncathan) [inline code/modules/atmospherics/gasmixtures/gas_mixture.dm]: STOP CHANGING NAMES AND COMMENTS TO BE WRONG AAAAA

File: code/modules/atmospherics/gasmixtures/gas_mixture.dm [hunk 4]
```diff
@@ -101,40 +99,38 @@ var/list/cached_gases_list = null
 		. += cached_gases[id][MOLES]
 
 /datum/gas_mixture/proc/return_pressure()
-	if(volume>0)
-		return total_moles()*R_IDEAL_GAS_EQUATION*temperature/volume
-	return 0
+	. = 0
+	if(volume > 0)
+		. = total_moles() * R_IDEAL_GAS_EQUATION * temperature / volume
```
Comment (duncathan) [inline code/modules/atmospherics/gasmixtures/gas_mixture.dm]: Bad

File: code/modules/atmospherics/gasmixtures/gas_mixture.dm [hunk 5]
```diff
@@ -8,91 +8,89 @@ What are the archived variables for?
...
 
-/datum/gas_mixture/New(Volume = CELL_VOLUME)
-	. = ..()
+/datum/gas_mixture/New(vol = CELL_VOLUME)
+	..()
 	gases = new
 	temperature = 0
 	temperature_archived = 0
-	volume = Volume
+	volume = 0
```
Comment (duncathan) [inline code/modules/atmospherics/gasmixtures/gas_mixture.dm]: uhhhh....

**Pull request merged**